paddle/fluid/framework/channel_test.cc

```diff
@@ -542,3 +542,341 @@ TEST(ChannelHolder, ChannelHolderUnBufferedSendReceiveTest) {
   ChannelHolderSendReceive(ch);
   delete ch;
 }
+
+TEST(ChannelHolder, ChannelUninitializedTest) {
+  ChannelHolder *ch = new ChannelHolder();
+  EXPECT_EQ(ch->IsInitialized(), false);
+  int i = 10;
+  EXPECT_EQ(ch->Send(&i), false);
+  EXPECT_EQ(ch->Receive(&i), false);
+  bool is_exception = false;
+  try {
+    ch->Type();
+  } catch (paddle::platform::EnforceNotMet e) {
+    is_exception = true;
+  }
+  EXPECT_EQ(is_exception, true);
+  delete ch;
+}
+
+TEST(ChannelHolder, ChannelInitializedTest) {
+  ChannelHolder *ch = new ChannelHolder();
+  ch->Reset<int>(2);
+  EXPECT_EQ(ch->IsInitialized(), true);
+  // Channel should remain intialized even after close
+  ch->close();
+  EXPECT_EQ(ch->IsInitialized(), true);
+  delete ch;
+}
+
+TEST(ChannelHolder, TypeMismatchSendTest) {
```

Comment:
Checking type matching or not isn't the function of Channel, is it necessary to add the unit test?

Reply:
Channel enforces the datatype in send (https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/channel.h#L74) and receive (https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/channel.h#L81)
 
The channel makes sure that the variable being sent or received is same as the type of the channel. The unit test is to make sure the paddle enforce is executed.

Reply:
I see, thanks!

Reply:
I see, thanks!